packages/now-client/src/create-deployment.ts

```diff
@@ -1,205 +1,173 @@
-      };
-    }
-
-    const files = await hashes(fileList);
-
-    debug(`Yielding a 'hashes-calculated' event with ${files.size} hashes`);
-    yield { type: 'hashes-calculated', payload: mapToObject(files) };
-
-    if (clientOptions.apiUrl) {
-      debug(`Using provided API URL: ${clientOptions.apiUrl}`);
-    }
-
-    if (clientOptions.userAgent) {
-      debug(`Using provided user agent: ${clientOptions.userAgent}`);
-    }
-
-    debug(`Setting platform version to ${version}`);
-    deploymentOptions.version = version;
-
-    debug(`Creating the deployment and starting upload...`);
-    for await (const event of upload(
-      files,
-      nowConfig,
-      clientOptions,
-      deploymentOptions
-    )) {
-      debug(`Yielding a '${event.type}' event`);
-      yield event;
     }
-  };
+  } else if (!isAbsolute(path)) {
+    throw new DeploymentError({
+      code: 'invalid_path',
+      message: `Provided path ${path} is not absolute`,
+    });
+  }
+
+  if (clientOptions.isDirectory && !Array.isArray(path)) {
+    debug(`Provided 'path' is a directory. Reading subpaths... `);
+    rootFiles = await readRootFolder(path);
+    debug(`Read ${rootFiles.length} subpaths`);
+  } else if (Array.isArray(path)) {
+    debug(`Provided 'path' is an array of file paths`);
+    rootFiles = path;
+  } else {
+    debug(`Provided 'path' is a single file`);
+    rootFiles = [path];
+  }
+
+  // Get .nowignore
+  let { ig, ignores } = await getNowIgnore(path);
+
+  debug(`Found ${ig.ignores.length} rules in .nowignore`);
+
+  let fileList: string[];
+
+  debug('Building file tree...');
+
+  if (clientOptions.isDirectory && !Array.isArray(path)) {
+    // Directory path
+    const dirContents = await readdir(path, ignores);
+    const relativeFileList = dirContents.map(filePath =>
+      relative(process.cwd(), filePath)
+    );
+    fileList = ig
+      .filter(relativeFileList)
+      .map((relativePath: string) => join(process.cwd(), relativePath));
+
+    debug(`Read ${fileList.length} files in the specified directory`);
+  } else if (Array.isArray(path)) {
+    // Array of file paths
+    fileList = path;
+    debug(`Assigned ${fileList.length} files provided explicitly`);
+  } else {
+    // Single file
+    fileList = [path];
+    debug(`Deploying the provided path as single file`);
+  }
+
+  if (!nowConfig) {
+    // If the user did not provide a nowConfig,
+    // then use the now.json file in the root.
+    const fileName = 'now.json';
+    const absolutePath = fileList.find(f => relative(cwd, f) === fileName);
+    debug(absolutePath ? `Found ${fileName}` : `Missing ${fileName}`);
+    nowConfig = await parseNowJSON(absolutePath);
+  }
+
+  if (nowConfig) {
+    // We'll apply all properties from `now.json`
+    // except for some to the payload.
+    const include = Object.assign({}, nowConfig);
+
+    delete include.scope;
+    delete include.github;
+
+    Object.assign(deploymentOptions, include);
```

Comment:
Nit pick: you can avoid looping twice and assign in one loop:

```ts
for (const [key, value] of Object.entries(nowConfig)) {
  if (key !== 'scope' && key !== 'github') {
    deploymentOptions[key] = value;
  }
}
```

Reply:
Yup, TS complains